source/backend/dal/Repositories/ResearchFileRepository.cs

```diff
@@ -138,6 +139,120 @@ public PimsResearchFile Update(PimsResearchFile researchFile)
             return researchFile;
         }
 
+        /// <summary>
+        /// Retrieves the research file with the specified id last update information.
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        public LastUpdatedByModel GetLastUpdateBy(long id)
+        {
+            var lastUpdatedByAggregate = new List<LastUpdatedByModel>();
+            var fileLastUpdatedBy = this.Context.PimsResearchFiles.AsNoTracking()
+                .Where(r => r.ResearchFileId == id)
+                .Select(r => new LastUpdatedByModel()
+                {
+                    ParentId = id,
+                    AppLastUpdateUserid = r.AppLastUpdateUserid,
+                    AppLastUpdateUserGuid = r.AppLastUpdateUserGuid,
+                    AppLastUpdateTimestamp = r.AppLastUpdateTimestamp,
+                })
+                .OrderByDescending(lu => lu.AppLastUpdateTimestamp)
+                .Take(1)
+                .ToList();
+            lastUpdatedByAggregate.AddRange(fileLastUpdatedBy);
+
+            var documentsLastUpdatedBy = this.Context.PimsResearchFileDocuments.AsNoTracking()
+                .Where(rd => rd.ResearchFileId == id)
+                .Include(rd => rd.Document)
+                .Select(rd => new LastUpdatedByModel()
+                {
+                    ParentId = id,
+                    AppLastUpdateUserid = rd.Document.AppLastUpdateUserid,
+                    AppLastUpdateUserGuid = rd.Document.AppLastUpdateUserGuid,
+                    AppLastUpdateTimestamp = rd.Document.AppLastUpdateTimestamp,
+                })
+                .OrderByDescending(lu => lu.AppLastUpdateTimestamp)
+                .Take(1)
+                .ToList();
+            lastUpdatedByAggregate.AddRange(documentsLastUpdatedBy);
+
+            // This is needed to get the document last-updated-by from the document that where deleted
+            var documentsHistoryLastUpdatedBy = this.Context.PimsResearchFileDocumentHists.AsNoTracking()
```

Comment:
From practical testing this doesn't seem to work, not sure why yet.

Reply:
(tried deleting a document, last update date/time does not change).

Reply:
Ok, think I've got it figured out. Backend logic works fine, frontend logic does not. Looks like the frontend is re-querying the update info before the actual delete is fully processed? if I do the delete the request doesn't show the correct time, but if I then refresh the page everything works.

Reply:
uhm, thats odd. Does the same happen with notes?